src/legacy/core_plugins/metrics/public/components/panel_config/gauge.js

```diff
@@ -143,10 +155,16 @@ class GaugePanelConfigUi extends Component {
                   />)}
                   fullWidth
                 >
-                  <EuiFieldText
-                    onChange={handleTextChange('filter')}
-                    value={model.filter}
-                    fullWidth
+                  <QueryBarInput
+                    query={{
+                      language: model.filter.language || uiSettingsQueryLanguage,
+                      query: model.filter.query || '',
+                    }}
+                    screenTitle={'GaugePanelConfigQuery'}
+                    onChange={this.handleQueryChange}
+                    appName={'VisEditor'}
+                    indexPatterns={[model.index_pattern || model.default_index_pattern]}
+                    store={localStorage || {}}
```

Comment:
I think it's better to see an exception if for some reason the  `localStorage  `is undefined. 

Probably exception should be raised from QueryBarInput component.